test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -8,6 +8,517 @@ test.describe( 'Links', () => {
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type(
+			'This is Gutenberg: https://wordpress.org/gutenberg'
+		);
+
+		// Select the URL.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+
+		// Click on the Link button.
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// A link with the selected URL as its href should have been inserted.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content:
+						'This is Gutenberg: <a href="https://wordpress.org/gutenberg">https://wordpress.org/gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `is not created when we click away from the link input`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		// Create a block with some text.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is Gutenberg' );
+
+		// Select some text.
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+
+		// Click on the Link button.
+		await page.getByRole( 'button', { name: 'Link' } ).click();
+
+		// Type a URL.
+		await page.keyboard.type( 'https://wordpress.org/gutenberg' );
+
+		// Click somewhere else - it doesn't really matter where.
+		await editor.canvas
+			.getByRole( 'textbox', { name: 'Add title' } )
+			.focus();
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is Gutenberg',
+				},
+			},
+		] );
+	} );
+
+	test( `can be edited`, async ( { page, editor, pageUtils, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Click on the Edit button.
+		await page.getByRole( 'button', { name: 'Edit' } ).click();
+
+		// Change the URL.
+		await page.getByPlaceholder( 'Search or type url' ).fill( '' );
+		await page.keyboard.type( '/handbook' );
+
+		// Submit the link.
+		await pageUtils.pressKeys( 'Enter' );
+
+		// The link should have been updated.
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is <a href="/handbook">Gutenberg</a>',
+				},
+			},
+		] );
+	} );
+
+	test( `can be removed`, async ( { page, editor, LinkUtils } ) => {
+		await LinkUtils.createAndReselectLink();
+
+		// Unlick via shortcut
+		// we do this to avoid an layout edge case whereby
+		// the rich link preview popover will obscure the block toolbar
+		// under very specific circumstances and screensizes.
+		await page.getByRole( 'button', { name: 'Unlink' } ).nth( 1 ).click();
```

Comment:
Why the `nth`? This seems like it could be very brittle.

Reply:
because the first one is the one in the block toolbar, I changed the selector to use the parent to specify which one we want to click, does that look better to you?